resources/testdata/data/json/dates/gregory@1/ar-EG.json

```diff
@@ -123,10 +123,250 @@
       "short": "h:mm a"
     },
     "date_time": {
-      "full": "{1} في {0}",
-      "long": "{1} في {0}",
-      "medium": "{1}, {0}",
-      "short": "{1}, {0}"
+      "style_patterns": {
+        "full": "{1} في {0}",
+        "long": "{1} في {0}",
+        "medium": "{1}, {0}",
+        "short": "{1}, {0}"
+      },
+      "available_formats": [
```

Comment:
I was trying to decide on the best representation for this data, as it's a key pair value. Making it a simple tuple seemed like the best choice here to me.

Ultimately, the key will be converted from `Cow<'static, str>` to a `struct Skeleton` that can then be used for skeleton matching. I'm not 100% sure if it's useful to provide these as `Cow`s, or what the ultimate design of the memory management for providers is. However, this is following the existing examples on how this data is being read in, so I'm assuming it's compatible.

My understanding of the data flow is
* The serialized JSON is read in.
* This gets deserialized into the `DateTimeFormatsV1` structure.
* `Skeleton`s are generated from the `DateTimeFormatsV1`
* Skeleton matching happens with the `Skeleton` struct.

Reply:
Yeah, I think that for now we can keep the key as `Cow<str>`, and separately once we have more complete data structures we can experiment with shifting from stringified representations to serializing/deserializing structs and see what it does to perf/mem/size.